posts/openscanhub-prototype/index.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "SAST using OpenScanHub is here!"
+date: 2024-07-30T08:46:18+02:00
+authors: lbarczio
+tags:
+  - copr
+  - srpm
+  - sast
+  - openscanhub
+---
+
+We are excited to announce a new experimental feature in our service: the integration of Static
+Application Security Testing (SAST) using [OpenScanHub](https://openscanhub.fedoraproject.org/).
+
+OpenScanHub is a service that runs various static analyzers on RPM packages - by default `Cppcheck`,
+`ShellCheck`, the static analyzers embedded in `GCC` and `Clang`, and the `find-unicode-control` tool.
+
+Let's have a look at the functionality we are bringing with this!
+
+<!--truncate-->
+
+## Functionality
+
+This functionality is designed to enhance software security and reliability by automatically
+scanning for vulnerabilities and issues in code submitted through pull requests.
+It is enabled using the configuration option
+[`osh_diff_scan_after_copr_build`](/docs/configuration#osh_diff_scan_after_copr_build),
+which is currently set to `true` by default. This option enables differential scanning
+in OpenScanHub for Copr builds triggered by pull requests, given that there is also a corresponding
+Copr build job with `trigger: commit` in the configuration with configured branch
+matching to the target branch of the pull request. This setup helps Packit
+obtain the base build necessary for differential scanning, resulting in more accurate
+detection of issues introduced by the pull request.
+
+To enable this feature, ensure you have Copr builds for commits configured
+(and both Copr builds for PRs and commits run for `fedora-rawhide-x86_64`).
+If not, you can add a job into `jobs` section in the Packit configuration as shown below:
+
+```yaml
+- job: copr_build
+  trigger: commit
+  branch: main
+  targets:
+    - fedora-all
+```
+
+This configuration assumes that pull requests are typically merged into the `main` branch.
+It can be further tweaked to preserve the Copr project (and not create a temporary one,
+see more details [here](https://packit.dev/docs/configuration/upstream/copr_build#optional-parameters))
+or use a custom Copr project in your Copr namespace (more details
+[here](https://packit.dev/docs/configuration/upstream/copr_build#using-a-custom-copr-project)).
+
+Here is an example of a scan showing some new findings:
+
+![Example findings](openscanhub-findings.png)
+
+## Conclusion
+
+As this is an initial prototype, the configuration and functionality may evolve based on user feedback
+(e.g. reporting the actual results of a scan).
+We invite you to try out this feature and [share your thoughts](/#contact) with us.
```

Comment:
Maybe open up a discussion thread [here](https://github.com/packit/packit-service/discussions) and link it? Especially since the mail is screwed :D

You can take the GitHub Merge Queue [discussion](https://github.com/orgs/community/discussions/46757) as an inspiration, they even list known issues there (maybe it's a good place for mentioning the issue with dynamic deps).

Reply:
good point, will do